### PR TITLE
Rotate a body

### PR DESCRIPTION
**Pull request by jonahrb**

### CODE REVIEW

File: tests/integration/test_design.py [hunk 1]
```diff
@@ -777,6 +777,23 @@ def test_bodies_translation(modeler: Modeler):
...
+    """Test for verifying the correct rotation of a ``Body``."""
+
+    # Create your design on the server side
+    design = modeler.create_design("BodyRotation_Test")
+
+    body = design.extrude_sketch("box", Sketch().box(Point2D([0, 0]), 1, 1), 1)
+    copy = body.copy(design, "copy")
+    body.rotate(Point3D([0, 0, 0]), UnitVector3D([0, 0, 1]), np.pi / 4)
+
+    # If rotated, uniting body and its copy should now result in a volume > 1
```
Comment (RobPasMue) [inline tests/integration/test_design.py:790]: Is it possible to perform some more checks on the Linux side?

File: tests/integration/test_design.py [hunk 2]
```diff
@@ -777,6 +777,23 @@ def test_bodies_translation(modeler: Modeler):
...
+    # boolean operations not on linux?
+    if modeler.client.backend_type is not BackendType.LINUX_SERVICE:
+        body.unite(copy)
+        assert body.volume.m > 1
+
+
```
Comment (RobPasMue) [inline tests/integration/test_design.py:796]: Nice way of verifying the rotation 😄

File: src/ansys/geometry/core/designer/body.py [hunk 3]
```diff
@@ -740,6 +767,27 @@ def translate(
...
+        rotation_magnitude = angle.value.m_as(DEFAULT_UNITS.SERVER_ANGLE)
+        self._grpc_client.log.debug(f"Rotating body {self.id}.")
+        self._bodies_stub.Rotate(
+            RotateRequest(
+                id=self.id,
+                axis_origin=point3d_to_grpc_point(axis_origin),
+                axis_direction=unit_vector_to_grpc_direction(axis_direction),
+                angle=rotation_magnitude,
+            )
+        )
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:789]: Is there no response coming from the server side?

Comment (RobPasMue) [reply src/ansys/geometry/core/designer/body.py:789]: A true/false message should be enough to see whether the body was effectively rotated I guess.

Comment (jonahrb) [reply src/ansys/geometry/core/designer/body.py:789]: @RobPasMue `rotate()` is basically identical to `translate()` in function, so I wanted to stay consistent between the 2 in design

File: tests/integration/test_design.py (hunk 1 above)
Comment (jonahrb) [reply tests/integration/test_design.py]: @RobPasMue just pushed new changes to the test that are cross-platform :)

**Review by RobPasMue — DISMISSED** [review]

**Comment by RobPasMue** [comment]
Looks like the Linux side of things is failing

**Comment by jonahrb** [comment]
> Looks like the Linux side of things is failing

@RobPasMue yeah not sure what's going on there, looks like it's not importing RotateRequest, which doesn't make sense

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**